dashboard/test/models/user_test.rb

```diff
@@ -1683,4 +1683,48 @@ def mock_geocoder_result(result)
     assert user.update(email: 'valid@example.net')
     refute user.update(email: 'invalid@incomplete')
   end
+
+  test 'find_or_create_teacher creates new teacher' do
+    admin = create :admin
+
+    params = {
+      email: 'email@example.net',
+      name: 'test user'
+    }
+
+    assert_creates(User) do
```

Comment:
Any reason not to do `teacher = assert_creates(User)...` so that you don't need to keep doing `User.last`?